src/Aspire.Hosting/ParameterResourceBuilderExtensions.cs

```diff
@@ -58,13 +61,39 @@ private static string GetParameterValue(IConfiguration configuration, string nam
         }
         catch (DistributedApplicationException ex)
         {
-            state = state with { State = "FailedToStart", Properties = [.. state.Properties, new("Value", ex.Message)] };
+            state = state with
+            {
+                State = new ResourceStateSnapshot("Configuration missing", KnownResourceStateStyles.Error),
+                Properties = [.. state.Properties, new("Value", ex.Message)]
+            };
+
+            builder.Services.AddLifecycleHook((sp) => new WriteParameterLogsHook(
+                sp.GetRequiredService<ResourceLoggerService>(),
+                name,
+                ex.Message));
         }
 
         return builder.AddResource(resource)
                       .WithInitialState(state);
     }
 
+    /// <summary>
+    /// Writes the message to the specified resource's logs.
+    /// </summary>
+    private sealed class WriteParameterLogsHook(ResourceLoggerService loggerService, string resourceName, string message) : IDistributedApplicationLifecycleHook
+    {
+        public Task AfterResourcesCreatedAsync(DistributedApplicationModel appModel, CancellationToken cancellationToken)
```

Comment:
This is too late as it runs after all resources are started.

Reply:
Too late for what?

We just want to log the exception message into the log for the resource.

Reply:
But you want it to show up quickly, not after all resources have started. If you're provisioning azure resources for e.g. this won't show up until after. These events run after all resources are provisioned.

Reply:
BeforeStartAsync works since this resource fails super early.